tests/unit/locations/history-fragment-test.ts

```diff
@@ -0,0 +1,143 @@
+import HistoryLocation from 'ember-osf-web/locations/history';
+import { setupTest } from 'ember-qunit';
+import { module, test } from 'qunit';
+import sinon from 'sinon';
+
+interface HistoryState {
+    path: string;
+    uuid: string;
+}
+
+class FakeHistory {
+    state: HistoryState | null = null;
+    private states: HistoryState[] = [];
+
+    replaceState(state: HistoryState) {
+        this.state = state;
+        this.states[0] = state;
+    }
+
+    pushState(state: HistoryState) {
+        this.state = state;
+        this.states.unshift(state);
+    }
+}
+
+function mockBrowserLocation(path: string): typeof window.location {
+    // This is a neat trick to auto-magically extract the hostname from any
+    // url by letting the browser do the work ;)
+    const tmp = document.createElement('a');
+    tmp.href = path;
+
+    const protocol = !tmp.protocol || tmp.protocol === ':' ? 'http' : tmp.protocol;
+    const pathname = tmp.pathname.match(/^\//) ? tmp.pathname : `/${tmp.pathname}`;
+
+    // @ts-ignore - mock just enough for HistoryLocation.getURL
+    return {
+        hash: tmp.hash,
+        host: tmp.host || 'localhost',
+        hostname: tmp.hostname || 'localhost',
+        href: tmp.href,
+        pathname,
+        port: tmp.port || '',
+        protocol,
+        search: tmp.search,
+    };
```

Comment:
Would be better to cast to get type safety:
```suggestion
    } as typeof window.location;
```

Reply:
ahh, nice, that's much better